Documentation for Salesforce DLS (#2056)
elastic/connectors: docs/document-level-security/SALESFORCE.md

### Setting up the Salesforce connector

See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.

## Document level security

Document level security (DLS) enables you to restrict access to documents based on a user'­s permissions. This feature is available by default for the Salesforce connector.
Salesforce connector DLS supports for both standard & custom objects.

Refer to [document level security](https://www.elastic.co/guide/en/enterprise-search/master/dls.html) for more information.

Salesforce allows users to set permissions in different ways i.e. via Profiles, Permission sets and Permission set Groups.

> Refer this tutorial to get more idea on setting the permissions - [link](https://howtovideos.hubs.vidyard.com/watch/B1bQnMFg2VyZq7V6zXQjPg#:~:text=This%20is%20a%20must%20watch,records%20in%20your%20Salesforce%20organization.)

Reviewer: ```suggestion
Refer to this tutorial to get more idea on setting the permissions - [link](https://howtovideos.hubs.vidyard.com/watch/B1bQnMFg2VyZq7V6zXQjPg#:~:text=This%20is%20a%20must%20watch,records%20in%20your%20Salesforce%20organization.)
```

> To ingest any standard or custom objects, users must ensure that at least `Read` permission is granted to that object through any of the following methods for setting permissions.

Reviewer: ```suggestion
To ingest any standard or custom objects, users must ensure that at least `Read` permission is granted to that object. This can be granted using any of the following methods for setting permissions.
```

### Set Permissions using Profiles

> Refer the [official documentation](https://help.salesforce.com/s/articleView?id=sf.admin_userprofiles.htm&type=5) to know how to set permissions via Profiles.

Reviewer: ```suggestion
Refer to the [official documentation](https://help.salesforce.com/s/articleView?id=sf.admin_userprofiles.htm&type=5) for setting permissions via Profiles.
```

### Set Permissions using Permissions Set

> Refer the [official documentation](https://help.salesforce.com/s/articleView?id=sf.perm_sets_overview.htm&language=en_US&type=5) to know how to set permissions via Permissions Sets.

Reviewer: ```suggestion
Refer to the [official documentation](https://help.salesforce.com/s/articleView?id=sf.perm_sets_overview.htm&language=en_US&type=5) for setting permissions via Permissions Sets.
```

### Set Permissions using Permissions Set group

> Refer the [official documentation](https://help.salesforce.com/s/articleView?id=sf.perm_set_groups.htm&type=5) to know how to set permissions via Permissions Set Groups.

Reviewer: ```suggestion
Refer to the [official documentation](https://help.salesforce.com/s/articleView?id=sf.perm_set_groups.htm&type=5) for setting permissions via Permissions Set Groups.
```

### Set Profiles, Permission Set and Permission Set Groups to the User

> Go to the `Administration` section => under `Users` section, select `Users` and choose the user to set the permissions to. Now, we can set the Profile, Permission Set and Permission Set Groups created in above steps.

Reviewer: ```suggestion
1. Go to the `Administration` section => under `Users` section
2. Select `Users` and choose the user to set the permissions to. 
3. Set the Profile, Permission Set or Permission Set Groups created in the earlier steps.
```

> **Note:** Refer to [DLS in Search Applications](https://www.elastic.co/guide/en/enterprise-search/master/dls-e2e-guide.html) to learn how to ingest data from Salesforce with DLS enabled, when building a search application.

Reviewer: ```suggestion
**Note:** Refer to [DLS in Search Applications](https://www.elastic.co/guide/en/enterprise-search/master/dls-e2e-guide.html) to learn how to ingest data with DLS enabled, when building a search application.
```

FYI that guide uses Sharepoint Online.

### Setting up the Salesforce connector

See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.

## Document level security

Document level security (DLS) enables you to restrict access to documents based on a user'­s permissions. This feature is available by default for the Salesforce connector.
Salesforce connector DLS supports for both standard & custom objects.

> Refer to [document level security](https://www.elastic.co/guide/en/enterprise-search/master/dls.html) for more information.

Reviewer 2: Should this link to /current/ ?

Author: I took the reference from already merged connector DLS documentation. Do we need to update this link?

Reviewer: We're on main here so master in the main docs makes sense

Salesforce allows users to set permissions in different ways i.e. via Profiles, Permission sets and Permission set Groups. 

Refer this tutorial to get more idea on setting the permissions - [link](https://howtovideos.hubs.vidyard.com/watch/B1bQnMFg2VyZq7V6zXQjPg#:~:text=This%20is%20a%20must%20watch,records%20in%20your%20Salesforce%20organization.)

> ### Set Permissions using Profiles

Author: Yes, the Read permission is sufficient for the users to ingest the objects.

Follow below steps to set permissions via Profiles:
1. From the setup page, go to `Administration` section => `Users` => `Profiles` and create a new profile
2. Choose `Read Only` or `Standard User` for the Existing Profile dropdown, give a name to the profile and save it. By default, `Read Only` or `Standard User` have the read permission to access all standard objects. There can be some more profiles which do have a read access to standard objects but these are some of them.
3. Now, edit the newly created profile and under `Object Permissions`, assign at least a `Read` access to the standard objects and custom objects you want to ingest into ElasticSearch.

> **Note:** If users specify advanced sync rules then they need to assign a `Read` access for that specific object in the profile.

Reviewer 2: > need to assign a `Read` access for that specific object

Is this true for both standard and custom objects?

Author: Yes, this is applicable to both standard and custom objects
